pageserver/src/tenant/timeline/detach_ancestor.rs

```diff
@@ -352,10 +358,72 @@ pub(super) async fn prepare(
 
     // TODO: copying and lsn prefix copying could be done at the same time with a single fsync after
     let mut new_layers: Vec<Layer> =
-        Vec::with_capacity(straddling_branchpoint.len() + rest_of_historic.len());
+        Vec::with_capacity(straddling_branchpoint.len() + rest_of_historic.len() + 1);
 
     {
-        tracing::debug!(to_rewrite = %straddling_branchpoint.len(), "copying prefix of delta layers");
+        tracing::info!("removing non-inherited keys by writing an image layer at the detach LSN");
```

Comment:
I wonder if it would make sense to move it to a different function. the function is already quite large.

Reply:
moved to a separate one